w3af/plugins/grep/cdn_providers.py

```diff
@@ -0,0 +1,255 @@
+        [".fastly.net", "Fastly"],
+        [".fastlylb.net", "Fastly"],
+        [".nocookie.net", "Fastly"],
+        [".gslb.taobao.com", "Taobao"],
+        [".gslb.tbcache.com", "Alimama"],
+        [".mirror-image.net", "Mirror Image"],
+        [".yottaa.net", "Yottaa"],
+        [".cubecdn.net", "cubeCDN"],
+        [".cdn77.net", "CDN77"],
+        [".cdn77.org", "CDN77"],
+        [".incapdns.net", "Incapsula"],
+        [".bitgravity.com", "BitGravity"],
+        [".r.worldcdn.net", "OnApp"],
+        [".r.worldssl.net", "OnApp"],
+        ["tbcdn.cn", "Taobao"],
+        [".taobaocdn.com", "Taobao"],
+        [".ngenix.net", "NGENIX"],
+        [".pagerain.net", "PageRain"],
+        [".ccgslb.com", "ChinaCache"],
+        ["cdn.sfr.net", "SFR"],
+        [".azioncdn.net", "Azion"],
+        [".azioncdn.com", "Azion"],
+        [".azion.net", "Azion"],
+        [".cdncloud.net.au", "MediaCloud"],
+        [".rncdn1.com", "Reflected Networks"],
+        [".cdnsun.net", "CDNsun"],
+        [".mncdn.com", "Medianova"],
+        [".mncdn.net", "Medianova"],
+        [".mncdn.org", "Medianova"],
+        ["cdn.jsdelivr.net", "jsDelivr"],
+        [".nyiftw.net", "NYI FTW"],
+        [".nyiftw.com", "NYI FTW"],
+        [".resrc.it", "ReSRC.it"],
+        [".zenedge.net", "Zenedge"],
+        [".lswcdn.net", "LeaseWeb CDN"],
+        [".lswcdn.eu", "LeaseWeb CDN"],
+        [".revcn.net", "Rev Software"],
+        [".revdn.net", "Rev Software"],
+        [".caspowa.com", "Caspowa"],
+        [".twimg.com", "Twitter"],
+        [".facebook.com", "Facebook"],
+        [".facebook.net", "Facebook"],
+        [".fbcdn.net", "Facebook"],
+        [".cdninstagram.com", "Facebook"],
+        [".rlcdn.com", "Reapleaf"],
+        [".wp.com", "WordPress"],
+        [".aads1.net", "Aryaka"],
+        [".aads-cn.net", "Aryaka"],
+        [".aads-cng.net", "Aryaka"],
+        [".squixa.net", "section.io"],
+        [".bisongrid.net", "Bison Grid"],
+        [".cdn.gocache.net", "GoCache"],
+        [".hiberniacdn.com", "HiberniaCDN"],
+        [".cdntel.net", "Telenor"],
+        [".raxcdn.com", "Rackspace"],
+        [".unicorncdn.net", "UnicornCDN"],
+        [".optimalcdn.com", "Optimal CDN"],
+        [".kinxcdn.com", "KINX CDN"],
+        [".kinxcdn.net", "KINX CDN"],
+        [".stackpathdns.com", "StackPath"],
+        [".hosting4cdn.com", "Hosting4CDN"],
+        [".netlify.com", "Netlify"],
+        [".b-cdn.net", "BunnyCDN"],
+        [".pix-cdn.org", "Advanced Hosters CDN"],
+        [".roast.io", "Roast.io"],
+        [".cdnvideo.ru", "CDNvideo"],
+        [".cdnvideo.net", "CDNvideo"],
+        [".trbcdn.ru", "TRBCDN"],
+        [".cedexis.net", "Cedexis"],
+        [".streamprovider.net", "Rocket CDN"],
+    ]
+
+    def grep(self, request, response):
+        """
+        Check if responses are hosted by CDN provider
+        """
+        headers = response.get_headers()
+        for cdn_header in self.cdn_headers:
```

Comment:
`for cdn_header in self.cdn_headers:` would look much better if you would use:

```python
for header_name, header_value, provider_name in self.cdn_headers:
```

That way it is easier to follow the code, and `cdn_header[1]` is replaced with the much better `header_value`

Reply:
Good point, shame on me I didn't fix that before PR 😉